Extensions/IISWebAppDeploy/Src/Tasks/SqlDacpacDeploy/DeployToSqlServer.ps1

```diff
@@ -81,22 +75,88 @@ function GetScriptToRun
 
     if ($taskType -eq "dacpac")
     {
-        $invokeMain = "Execute-DacpacDeployment -dacpacFile `"$dacpacFile`" -targetMethod $targetMethod -serverName `"$serverName`" -databaseName `"$databaseName`" -authscheme $authscheme -sqlServerCredentials `$sqlServerCredentials -connectionString `"$connectionString`" -publishProfile `"$publishProfile`" -additionalArguments `"$additionalArguments`""
-
+        
         $sqlPackageScript = Get-Content TaskModuleSqlUtility\SqlPackageOnTargetMachines.ps1 | Out-String
 
+        try{
+            $sqlSplatArguments = @{
+                dacpacFile=$dacpacFile 
+                targetMethod=$targetMethod 
+                serverName=$serverName
+                databaseName=$databaseName
+                authscheme=$authscheme 
+                sqlServerCredentials="`$sqlServerCredentials"
+                connectionString=$connectionString 
+                publishProfile=$publishProfile 
+                additionalArguments=$additionalArguments
+            }
+            $jsonSqlSplatArguments = $sqlSplatArguments | ConvertTo-Json
+            $remoteArgsInit = "`$remoteJsonSqlSplattedArgs = '$jsonSqlSplatArguments'
```

Comment:
We are using splatting to prevent expansion of special character, there is not way directly to pass objects to remote machine for execution, So we need to serialize it to json and then on remote machine we are expanding it.